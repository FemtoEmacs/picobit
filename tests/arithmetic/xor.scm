;; bitwise xor tests

(define-macro (test-= lhs rhs name) ;; TODO watch out for name capture
  `(let ((DUMMY-lhs ,lhs)
	 (DUMMY-rhs ,rhs))
     (display ,name)
     (if (= DUMMY-lhs DUMMY-rhs)
	 (display " : PASSED\n")
	 (begin (display " : FAILED, got : ")
		(display DUMMY-lhs)
		(display " and ")
		(display DUMMY-rhs)
		(display "\n")))))

(test-= (bitwise-xor #xf0 #x0f) #xff "#xf0 | #x0f = #xff")
(test-= (bitwise-xor #x30 #x00) #x30 "#x30 | #x00 = #x30")
(test-= (bitwise-xor #x00 #x05) #x05 "#x00 | #x05 = #x05")
(test-= (bitwise-xor #x08 #x05) #x0D "#x08 | #x05 = #x0D")
(test-= (bitwise-xor #x18 #x05) #x1D "#x18 | #x05 = #x1D")
(test-= (bitwise-xor #x18 #x25) #x3D "#x18 | #x25 = #x3D")
(test-= (bitwise-xor #x18 #x35) #x2D "#x18 | #x35 = #x2D")
(test-= (bitwise-xor #xF7 #xDD) #x2A "#xF7 | #xDD = #x2A")
