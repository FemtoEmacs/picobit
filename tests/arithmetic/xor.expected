:0080000080
:10800000FBD7D30281764000818640008151400039
:10801000815B4000818440008190400081A140004C
:1080200081AD400081C9400081D1400081574000AE
:10803000816C400081394000813F4000811B40003D
:108040008118400081154000802701758034017738
:1080500080240178802701798024017A8036017B91
:108060008027017C8027017D804A017E8024017F5A
:1080700080490180807C0181807C0182804901836C
:10808000803C018580410187808001888042000218
:10809000807C01898048018A807C018B803E016A56
:1080A0008034018C8024018D8054018E8024018FC6
:1080B000806E0191807C0192807C0193807101949B
:1080C0008024019580240196807C0197806B019823
:1080D000808001998041019A8024019B8034019C19
:1080E0008027019D804A019E8037019F803601A034
:1080F000802701A2802701A3808001A4807201A5AE
:10810000802401A6803501A7803E01A8802401A912
:10811000807C01AA807301AB802401AC807701AE22
:10812000803401AF802401B0802401B1803001B2DD
:10813000803401B3803901B4800E0002807C01B527
:10814000808001B6803E01B7802701B8802701B941
:10815000802701BA80390002804101BB803C01BC0C
:10816000802401BD802401BE802401BF805701C04E
:10817000807C01C1807C01C2806701C3803401C45E
:10818000803C01C5807301C6802401C7807C01C882
:10819000807801CA802701CB802701CC802400028F
:1081A000808001CD802701CE803401CF804D01D069
:1081B000802401D2806A0002803701D3803601D446
:1081C000802701D5807C01D6807C01448024014534
:1081D000803901468027014780400148808001495D
:1081E000804800028040014A807C014B803B014C6A
:1081F0008041014D8048014E806A014F8037015017
:108200008048015180270152802401538024015469
:10821000802401558024015680340158802401595E
:108220008034015A806A015C8080015D8045015E76
:108230008027015F802701608069016180340162CD
:108240008035016380660164804101658027016695
:108250008035017B80730167802401688024016977
:108260008032016A8024016B807C016D8045016E43
:10827000803400028039016F80270170807C017199
:10828000807C0172802401738068016A80270174F8
:10829000803C012C8024012D8041012E8037017B00
:1082A0008066012F8027013080270131807D0132D7
:1082B00080390133808001348027013580240136E4
:1082C00080240137802401388034013A8024013B26
:1082D0008024013C807C013D807C013E807800024E
:1082E0008024014080240141806501428041014396
:1082F00080270121805701228034017B80350123B2
:1083000080780126802401248024012580700126A4
:108310008027012780480128802701298024012AFD
:10832000807C012B807C011A8024011C807C011D33
:108330008039011E8050011F8027012080410117D4
:108340008034011980450002807C0116806A01702A
:1083500002B405DF5002B405D951A0F413F6A1030D
:1083600002B405AE62EEA03404F6A03402B4058374
:1083700062EE0409F60902B4055862EE0C09F61122
:1083800002B4052D62EE1C09F6A02102B4050262BA
:10839000EE1CA029F6A04102B404D762EE1CA0395D
:1083A000F6A03102B404AC62EEA0FBA0E1F6A02E70
:1083B00002B403B762EED802A10FB58BEE2022C73C
:1083C000B30491A104B6800120DAB8954121DBB64F
:1083D000800221CC9A21CE2161EE2022CFB67200FC
:1083E000EF20B6800120DA9EA02607E620B558EEE1
:1083F000A02607E600EF20C0B8BF20B582B648012E
:108400002004C8B8AEA03121B5A402B583CDDCEFFD
:1084100002A034210EC5C122CD2102CDB4042071A9
:1084200001210EC89220EF210E406221B6620104A4
:1084300021C2EF2002B559B75520CCB8B8A02C07FF
:10844000E620CEB51FEE20CFB6800120D296A02D1B
:1084500007E600EF20CC9DA02407E620CEB505EE70
:1084600020CFB666A110B003C7EE20B003E4EEA0A3
:108470002D07E600EF20D995A112B103C720F09C8B
:108480002095A10EB103C7A10AB103C7A113B1037F
:10849000C7A105B003C7EE20B003C7EEA107B00324
:1084A000C7EE21B003C7EEA106B103C702A114B005
:1084B00003C7EE2022C795A104B103C7A105B003ED
:1084C000C7EE20B003C7EEA107B003C7EE21B0038B
:1084D000C7EEA106B103C702A10BB003C7EE20226D
:1084E000C795A104B103C7A105B003C7EE20B0032F
:1084F000C7EEA107B003C7EE21B003C7EEA106B1D6
:1085000003C702A109B003C7EE2022C795A104B199
:1085100003C7A105B003C7EE20B003C7EEA107B0A3
:1085200003C7EE21B003C7EEA106B103C702A10C39
:10853000B003C7EE2022C795A104B103C7A105B0BF
:1085400003C7EE20B003C7EEA107B003C7EE21B00A
:1085500003C7EEA106B103C702A108B003C7EE200E
:1085600022C795A104B103C7A105B003C7EE20B08F
:1085700003C7EEA107B003C7EE21B003C7EEA10603
:10858000B103C702A111B003C7EE2022C795A10411
:10859000B103C7A105B003C7EE20B003C7EEA10722
:1085A000B003C7EE21B003C7EEA106B103C702A115
:1085B0000DB003C7EE2022C795A104B103C7A105E2
:1085C000B003C7EE20B003C7EEA107B003C7EE218A
:1085D000B003C7EEA106B103C7012007E600EF0212
:1085E0002004C8212302CDCDB405EC71012204C8BA
:1085F00022242302CDCDCDB405FB710123B8DC21AB
:10860000B0042E23222602CDCDCDB4060E71012258
:10861000B8C621B0042E21242602CDCDCDB406212A
:1086200071012022242602CDCDCDCDB4062F7000BD
:108630002121C4232502CDCDB4063C710122B8937B
:10864000219220EF2298019420B1042E20EF21B82E
:0E8650007BB77500B86EB76A21B73B21B7251E
:00000001FF
